### PR TITLE
Set up a "Hello world" project as one app with Webpack

### PR DESCRIPTION
**Pull request by David-Kasilia**
- Create a new Rails app called 'hello-rails-react'.
      - Use the command `rails new -j webpack my_new_rails_app` as described in the previous lesson's tutorial.
 - Verify that the `jsbundling-rails` gem is added to your Gemfile.
 - Initialize your project with Git.
 - Install needed node dependencies as described in the previous lesson's tutorial.
 - Create a `babel.config.js` in the root folder of your project as described in the previous lesson's tutorial.
 - Update `webpack.config.js` to include the babel loader as described in the previous lesson's tutorial.
 - Make sure that your project has a Postgres database set up. 
 - Create a table for storing your messages and create 5 different greetings. 
 - Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
 - Create a static view that will be the root of your app.

---------

- Create the `App` component with react-router and render it in your static view.
- Create the `Greeting` component that will display a greeting. Set it up as a route in your App component.
- Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- Display the random greeting in your `Greeting` component.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,72 @@
+source 'https://rubygems.org'
```
Comment (gemmen29) [inline Gemfile:1]: - [x] Kindly fix this error 
image
when I run your project I get this error and nothing is displayed in the greeting route.
image

File: app/controllers/api/v1/messages_controller.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Api::V1::MessagesController < ApplicationController
+  def index
+    render json: Message.all
+  end
```
Comment (gemmen29) [inline app/controllers/api/v1/messages_controller.rb]: - [x] Kindly make the endpoint only return one random greeting message as per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/connect-frontend-frameworks/rails_react_webpack.md#:~:text=Create%20an%20API%20endpoint%20that%20selects%20a%20random%20greeting%20from%20your%20table).

File: app/controllers/api/v1/messages_controller.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class Api::V1::MessagesController < ApplicationController
+  def index
+    render json: Message.all
```
Comment (gemmen29) [inline app/controllers/api/v1/messages_controller.rb]: - [x] Kindly add to the `Message` table you created 5 different greeting messages as per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/connect-frontend-frameworks/rails_react_webpack.md#:~:text=Create%20a%20table%20for%20storing%20your%20messages%20and%20create%205%20different%20greetings.).
### Hint: you can use the `seeds.rb` for adding the greeting messages.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @David-Kasilia, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/api/v1/messages_controller.rb (hunk 3 above)
Comment (David-Kasilia) [reply app/controllers/api/v1/messages_controller.rb]: Hello @gemmen29 . Thank you for your review. The Table already has five messages I added on my side

File: Gemfile (hunk 1 above)
Comment (David-Kasilia) [reply Gemfile:1]: Hello @gemmen29 . Thank you for your review. The URL from my side is working well.

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @David-Kasilia, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights:
- All linters passing
- All requirements completed
- Professional readme

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**